crates/cairo-lint-core/src/lints/equal_operand.rs

```diff
@@ -0,0 +1,30 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{Expr, ExprBinary};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{TypedStablePtr, TypedSyntaxNode};
+
+pub const REDUNDANT_COMPARISON: &str = "redundant comparison found. This expression always evaluates to true or false.";
+
+pub fn check_redundant_comparison(db: &dyn SyntaxGroup, expr: &ExprBinary, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let lhs = expr.lhs(db);
+    let rhs = expr.rhs(db);
+
+    let is_function_or_method_call = |expr: &Expr| match expr {
+        Expr::FunctionCall(_) => true,
+        Expr::StructCtorCall(_) => true,
+        _ => false,
+    };
```

Comment:
hey @Gerson2102 is this

`pub fn is_function_or_method_call(db: &dyn SyntaxGroup, expr: &Expr) -> bool {
    let syntax_node = expr.as_syntax_node();
    match syntax_node.kind(db) {
        SyntaxKind::ExprFunctionCall | SyntaxKind::ExprStructCtorCall => true,
        _ => false,
    }
}
` 

the same approach as u have there? maybe checking it from `SyntaxKind` does the trick 🤔  

 `use cairo_lang_syntax::node::kind::SyntaxKind;`